homeassistant/components/slide/__init__.py

```diff
@@ -0,0 +1,159 @@
+"""Component for the Go Slide API."""
+
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_USERNAME,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_TIMEOUT,
+)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.helpers.event import async_track_time_interval, async_call_later
+from .const import (
+    DOMAIN,
+    SLIDES,
+    API,
+    COMPONENT,
+    DEFAULT_TIMEOUT,
+    CONF_RETRY,
+    DEFAULT_RETRY,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_SCAN_INTERVAL = timedelta(seconds=30)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_USERNAME): cv.string,
+                vol.Required(CONF_PASSWORD): cv.string,
+                vol.Optional(
+                    CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
+                ): cv.time_period,
+                vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
```

Comment:
Why do we need a config option for timeout?

Reply:
The timeout option is the aiohttp timeout for the cloud requests. The default is 300+ seconds, that's why i make it configurable (default is 30 seconds).

Reply:
Isn't it enough to always use th default?

Reply:
That's fine with me too, i will remove this in the next commit.

Reply:
Made the change, so removed.